packages/flutter/lib/src/gestures/tap.dart

```diff
@@ -167,28 +182,49 @@ class TapGestureRecognizer extends PrimaryPointerGestureRecognizer {
   bool _sentTapDown = false;
   bool _wonArenaForPrimaryPointer = false;
   Offset _finalPosition;
+  // The buttons sent by `PointerDownEvent`. If a `PointerMoveEvent` comes with a
+  // different set of buttons, the gesture is rejected and canceled.
+  int _initialButtons;
+
+  @override
+  bool isPointerAllowed(PointerDownEvent event) {
+    if (!isSingleButton(event.buttons)) {
+      return false;
+    }
+    return super.isPointerAllowed(event);
+  }
+
+  @override
+  void addAllowedPointer(PointerDownEvent event) {
+    super.addAllowedPointer(event);
+    // `_initialButtons` must be assigned here instead of `handlePrimaryPointer`,
+    // because `acceptGesture` might be called before `handlePrimaryPointer`,
+    // which relies on `_initialButtons` to create `TapDownDetails`.
+    _initialButtons = event.buttons;
+  }
 
   @override
   void handlePrimaryPointer(PointerEvent event) {
     if (event is PointerUpEvent) {
       _finalPosition = event.position;
-      if (_wonArenaForPrimaryPointer) {
-        resolve(GestureDisposition.accepted);
-        _checkUp();
-      }
+      _checkUp();
```

Comment:
So it's no longer resolves as "accepted" here? `_checkUp` doesn't seem to do that.

Reply:
(Still not explained)

Reply:
There are 2 changes in order to make logic clearer:
First, `_wonArenaForPrimaryPointer` is moved to `_checkUp`.

Note that
1. `_checkUp` is a no-op when `_wonArenaForPrimaryPointer = false`
2. `_wonArenaForPrimaryPointer` is set to `true` only when when the gesture is accepted.

Therefore, `resolve(GestureDisposition.accepted)` will never be run unless the gesture is already accepted, making it useless.

Reply:
Got it, makes sense.  Thanks for the explanation.